clients/src/main/java/org/apache/kafka/clients/admin/FenceProducersResult.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.clients.admin.internals.CoordinatorKey;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.annotation.InterfaceStability;
+import org.apache.kafka.common.utils.ProducerIdAndEpoch;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * The result of the {@link Admin#fenceProducers(Collection)} call.
+ *
+ * The API of this class is evolving, see {@link Admin} for details.
+ */
+@InterfaceStability.Evolving
+public class FenceProducersResult {
+
+    private final Map<CoordinatorKey, KafkaFuture<ProducerIdAndEpoch>> futures;
+
+    FenceProducersResult(Map<CoordinatorKey, KafkaFuture<ProducerIdAndEpoch>> futures) {
+        this.futures = futures;
+    }
+
+    /**
+     * Return a map from transactional ID to futures which can be used to check the status of
+     * individual fencings.
+     */
+    public Map<String, KafkaFuture<Void>> fencedProducers() {
```

Comment:
It feels like a bit of a shame that we didn't propose to a public version of `ProducerIdAndEpoch`, which would result in a more natural API than this `Void` future plus the methods for getting the PID and epoch for individual transactional ids.

Reply:
Yeah, that does seem more ergonomic... if it's any consolation I don't know that the PID/epoch are going to be tremendously useful for this API any time soon. We included them at this stage as a "just in case they come in handy later" measure but considering the point of this API is to block other producers from sending data and not really to gather metadata about transaction coordinator state, it shouldn't be a huge loss if it's a little tricky to get to that info.